pkg/graph/resolvers/system_profile_section_locks.go

```diff
@@ -0,0 +1,267 @@
+	systemLockStatusMap map[string]sectionLockStatusMap
+
+	sessionLockController struct {
+		systemLockStatuses systemLockStatusMap
+		sync.Mutex
+	}
+)
+
+var (
+	systemProfileSessionLocks sessionLockController
+)
+
+func init() {
+	systemProfileSessionLocks = sessionLockController{systemLockStatuses: make(systemLockStatusMap)}
+}
+
+// cedarSystemIDToSessionID converts a CEDAR system ID string to a UUID for pubsub sessions.
+// The same cedarSystemID always maps to the same sessionID for consistency.
+// TODO: Remove this conversion if/when cedarSystemId is migrated from string to UUID in the schema.
+func cedarSystemIDToSessionID(cedarSystemID string) uuid.UUID {
+	return uuid.NewSHA1(uuid.NameSpaceOID, []byte(cedarSystemID))
+}
+
+// GetSystemProfileSectionLocks returns the list of locked system profile sections. Any sections not included should be considered as unlocked.
+func GetSystemProfileSectionLocks(cedarSystemID string) ([]*models.SystemProfileSectionLockStatus, error) {
+	systemProfileSessionLocks.Lock()
+
+	sectionLocks, found := systemProfileSessionLocks.systemLockStatuses[cedarSystemID]
+	if !found {
+		systemProfileSessionLocks.Unlock()
+		return nil, nil
+	}
+
+	var lockStatuses []*models.SystemProfileSectionLockStatus
+	for section := range sectionLocks {
+		lockStatus := sectionLocks[section]
+		lockStatuses = append(lockStatuses, &lockStatus)
+	}
+	systemProfileSessionLocks.Unlock()
+
+	return lockStatuses, nil
+}
+
+// SubscribeSystemProfileSectionLockChanges creates a Subscriber and registers it for the pubsubevents.SystemProfileSectionLocksChanged event
+func SubscribeSystemProfileSectionLockChanges(ps pubsub.PubSub, cedarSystemID string, subscriber *subscribers.SystemProfileLockChangedSubscriber, onDisconnect <-chan struct{}) (<-chan *models.SystemProfileSectionLockStatusChanged, error) {
+	sessionID := cedarSystemIDToSessionID(cedarSystemID)
+	ps.Subscribe(sessionID, pubsubevents.SystemProfileSectionLocksChanged, subscriber, onDisconnect)
+	return subscriber.GetChannel(), nil
+}
+
+// LockSystemProfileSection will lock the provided system profile section on the provided system
+func LockSystemProfileSection(ps pubsub.PubSub, cedarSystemID string, section models.SystemProfileLockableSection, principal authentication.Principal) (bool, error) {
+	systemProfileSessionLocks.Lock()
+
+	sectionLocks, found := systemProfileSessionLocks.systemLockStatuses[cedarSystemID]
+	if !found {
+		systemProfileSessionLocks.systemLockStatuses[cedarSystemID] = make(sectionLockStatusMap)
+		sectionLocks = systemProfileSessionLocks.systemLockStatuses[cedarSystemID]
+	}
+
+	lockStatus, sectionWasLocked := sectionLocks[section]
+	if sectionWasLocked && lockStatus.LockedByUserAccount.ID != principal.Account().ID {
+		systemProfileSessionLocks.Unlock()
+		return false, fmt.Errorf("failed to lock section [%v], already locked by [%v]", section, lockStatus.LockedByUserAccount.ID)
+	}
+
+	account := principal.Account()
```

Comment:
Perhaps this can be moved higher in the stack? (the account will need to be referenced earlier for the id)